Framework/Core/include/Framework/InputRecord.h

```diff
@@ -29,11 +29,74 @@
 #include <memory>
 #include <type_traits>
 
+template <typename T>
+using default_delete = std::default_delete<T>;
+
 namespace o2 {
 namespace framework {
 
 struct InputSpec;
 
+/// A deleter type to be used with unique_ptr, which can be marked that
+/// it does not own the underlying resource and thus should not delete it.
+/// The resource ownership property controls the behavior and can only be
+/// set at construction of the deleter in the unique_ptr. Falls back to
+/// default_delete if not initialized to 'NotOwning'.
+/// Usage: unique_ptr<T, Deleter<T>> ptr(..., Deleter<T>(false))
+///
+/// By contract, the underlying, not owned resource is supposed to be
+/// available during the lifetime of the object, which is the case in the
+/// InputRecord and DPL processing APIs. The Deleter can be extended
+/// to support a callback to call a resource management outside.
+template <typename T>
+class Deleter : public default_delete<T>
```

Comment:
I would use a more explicative name than simply `Deleter`. Notice also that this effectively means we will be using an extra int per pointer. Is that reasonable ok?

Reply:
Agreed, how about moving `Deleter` into `InputRecord`, than it becomes `InputRecord::Deleter` which is probably explicative enough?

Regarding the increased size for a pointer I don't think its and issue as this is per incoming data blocks, definitely less than O(1k).

Reply:
Sounds good.